File: backend/src/main/java/com/plugga/backend/service/DeckService.java
package com.plugga.backend.service;

import com.plugga.backend.entity.Deck;

import java.util.List;

public interface DeckService {
    List<Deck> findAll();

    Deck findById(int id);

    void save(Deck deck);

    void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/service/DeckService.kt
package com.plugga.backend.service

import com.plugga.backend.entity.Deck
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface DeckService {
    fun findAll(pageable: Pageable): Page<Deck>

    fun findById(id: Int): Deck?

    fun findByUserId(pageable: Pageable, id: Int): Page<Deck>

    fun save(deck: Deck)

    fun deleteById(id: Int)
}
